test/core/plugin/sdk/funnelProtocol.test.js

```diff
@@ -13,28 +14,42 @@ const {
 
 describe('Test: sdk/funnelProtocol', () => {
   let request;
-  let funnel;
+  let kuzzle;
   let funnelProtocol;
 
   beforeEach(() => {
-    funnel = {};
+    kuzzle = new KuzzleMock();
 
-    funnel.executePluginRequest = sinon.stub().resolves('sdk result');
+    kuzzle.funnel.executePluginRequest = sinon.stub().resolves('sdk result');
 
     request = {
       controller: 'foo',
       action: 'bar'
     };
 
-    funnelProtocol = new FunnelProtocol(funnel);
+    funnelProtocol = new FunnelProtocol(kuzzle);
   });
 
   describe('#constructor', () => {
     it('should throw an InternalError if the funnel is instantiated without a valid User object', () => {
       should(() => {
-        new FunnelProtocol(funnel, { id: 42 });
+        new FunnelProtocol(kuzzle, { id: 42 });
       }).throw(PluginImplementationError, { id: 'plugin.context.invalid_user' });
     });
+
+    it('should forward messages received from the internalProtocol to the SDK', done => {
+      const payload = { room: 'room-id', hello: 'Gordon' };
+
+      funnelProtocol.on('room-id', () => {
+        should(funnelProtocol.kuzzle.on).be.calledOnce();
+        should(funnelProtocol.kuzzle.on.getCall(0).args[0])
+          .be.eql('core:network:internal:message');
+
+        done();
```

Comment:
If a `should` assertion fails, the callback won't be invoked, potentially making it painful to debug.

```suggestion
        try {
          should(funnelProtocol.kuzzle.on).be.calledOnce();
          should(funnelProtocol.kuzzle.on.getCall(0).args[0])
            .be.eql('core:network:internal:message');

          done();
        }
        catch (e) {
          done(e);
        }
```